PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/actions/SelectSubscriptionsAction.java

```diff
@@ -82,11 +81,12 @@ public void update(AnActionEvent e) {
         }
 
         final SubscriptionManager subscriptionManager = manager.getSubscriptionManager();
-
-        return loadSubscriptions(subscriptionManager, project)
-            .switchMap((subs) -> selectSubscriptions(project, subs))
-            .toSingle()
-            .doOnSuccess((subs) -> Optional.ofNullable(subs).ifPresent(subscriptionManager::setSubscriptionDetails));
+        List<SubscriptionDetail> subs = subscriptionManager.getSubscriptionDetails();
```

Comment:
you cannot convert async loadSubscriptions to sync in ui thread, this will block ui

Reply:
in fact the original codes is also blocking until the async task completes.

Reply:
AzureTaskManager.doRunAndWait

Reply:
reverted since the root cause is not here,